### PR TITLE
EQL: make endsWith function use a wildcard ES query wherever possible

### PR DESCRIPTION
**Pull request by astefan**
Addresses https://github.com/elastic/elasticsearch/issues/61154.

### CODE REVIEW

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-windows

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-windows

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-windows

File: x-pack/plugin/eql/src/test/resources/queryfolder_tests.txt [hunk 1]
```diff
@@ -105,13 +105,8 @@ process where cidrMatch(source_address, "10.0.0.0/8") != false
 twoFunctionsEqualsBooleanLiterals-caseSensitive
 process where endsWith(process_path, 'x') == true and endsWith(process_path, 'yx') == false
 ;
-"bool":{"must":[{"term":{"event.category":{"value":"process"
-"must":[{"script":{"script":{"source":"InternalQlScriptUtils.nullSafeFilter(
-InternalEqlScriptUtils.endsWith(InternalQlScriptUtils.docValue(doc,params.v0),params.v1,params.v2))","lang":"painless",
-"params":{"v0":"process_path","v1":"x","v2":true}}
-"script":{"script":{"source":"InternalQlScriptUtils.nullSafeFilter(InternalQlScriptUtils.not(
-InternalEqlScriptUtils.endsWith(InternalQlScriptUtils.docValue(doc,params.v0),params.v1,params.v2)))","lang":"painless",
-"params":{"v0":"process_path","v1":"yx","v2":true}}
```
Comment (costin) [inline x-pack/plugin/eql/src/test/resources/queryfolder_tests.txt:114]: Nice.

**Review by costin — APPROVED** [review]
Same comment as #61189

**Pull request merged**